src/features/customer/pages/AddAddress.tsx

```diff
@@ -0,0 +1,144 @@
+import { useForm, SubmitHandler } from "react-hook-form";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useNavigate } from "react-router-dom";
+import axiosInstance from "@/core/axiosInstance";
+
+interface AddAddressFormInputs {
+  streetName: string;
+  city: string;
+  apartment?: string;
+  province: string;
+  postalCode: string;
+  email: string;
+}
+
+export default function AddAddress() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<AddAddressFormInputs>();
+  const navigate = useNavigate();
+
+  const onSubmit: SubmitHandler<AddAddressFormInputs> = async (data) => {
+    try {
+      // Call the API to save the address
+      const res = await axiosInstance.post("/address", data);
+      console.log("Address saved:", res.data);
+      // Redirect back to the address list
+      navigate("/account/address");
```

Comment:
@tayojs you might need to add replace here so that when user click on back arrow. it doesnt come here.

As in:
navigate("/account/address", replace);